packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -1517,8 +1517,8 @@ class _RenderRangeSlider extends RenderBox with RelayoutWhenSystemFontsChangeMix
       isEnabled: isEnabled,
     );
 
-    final bool startThumbSelected = _lastThumbSelection == Thumb.start;
-    final bool endThumbSelected = _lastThumbSelection == Thumb.end;
+    final bool startThumbSelected = _lastThumbSelection == Thumb.start && !hoveringEndThumb;
```

Comment:
I see that `startThumbSelected` and `endThumbSelected` are also used in later code, deciding whether the thumb paint shows `isPressed` variant. I suspect that we don't want to change that. (Though there's no unit test to verify this...)

Anyway, unless it's intentional by you, I suggest moving these extra conditions to L1525 and L1541.

Reply:
@dkwingsmt If we paint `thumbShape` with previous condition, then shadow and other decoration effect will still be shown which is not intentional.

Reply:
So, I think current update is as expected. if you want i can update current test to verify that.

Reply:
@rkishan516 That would be awesome! Yeah can you add that?

Reply:
Sure

Reply:
I have updated test.